molgenis-data-platform/src/main/java/org/molgenis/data/platform/decorators/MolgenisRepositoryDecoratorFactory.java

```diff
@@ -95,8 +103,15 @@ public MolgenisRepositoryDecoratorFactory(EntityManager entityManager,
 		// 10. Register the cud action needed to index indexed repositories
 		decoratedRepository = new IndexActionRepositoryDecorator(decoratedRepository, indexActionRegisterService);
 
-		// 9. Custom decorators
-		decoratedRepository = repositoryDecoratorRegistry.decorate(decoratedRepository);
+		// 9. Custom & dynamic decorators
+		if (systemEntityTypeRegistry.hasSystemEntityType(repository.getEntityType().getId()))
```

Comment:
Fix this workaround for a bootstrapping issue. It should be possible to wrap ```SystemEntityType``` with ```DynamicDecorator```.